src/main/java/com/exasol/sql/expression/rendering/ValueExpressionRenderer.java

```diff
@@ -4,124 +4,234 @@
 import com.exasol.sql.ColumnsDefinition;
 import com.exasol.sql.UnnamedPlaceholder;
 import com.exasol.sql.expression.*;
+import com.exasol.sql.expression.comparison.Comparison;
+import com.exasol.sql.expression.comparison.ComparisonVisitor;
+import com.exasol.sql.expression.comparison.LikeComparison;
+import com.exasol.sql.expression.comparison.SimpleComparison;
+import com.exasol.sql.expression.function.AbstractFunction;
 import com.exasol.sql.expression.function.Function;
+import com.exasol.sql.expression.function.FunctionVisitor;
 import com.exasol.sql.expression.function.exasol.CastExasolFunction;
 import com.exasol.sql.expression.function.exasol.ExasolFunction;
 import com.exasol.sql.expression.function.exasol.ExasolUdf;
+import com.exasol.sql.expression.literal.*;
 import com.exasol.sql.rendering.ColumnsDefinitionRenderer;
 import com.exasol.sql.rendering.StringRendererConfig;
 
 /**
  * Renderer for common value expressions.
  */
-public class ValueExpressionRenderer extends AbstractExpressionRenderer implements ValueExpressionVisitor {
+public class ValueExpressionRenderer extends AbstractExpressionRenderer implements BooleanExpressionVisitor,
+        ComparisonVisitor, ValueExpressionVisitor, LiteralVisitor, FunctionVisitor {
+    int nestedLevel = 0;
+
+    /**
+     * Create a new instance of {@link ValueExpressionRenderer}.
+     * 
+     * @param config render configuration
+     */
     public ValueExpressionRenderer(final StringRendererConfig config) {
         super(config);
     }
 
     @Override
-    public void visit(final StringLiteral literal) {
-        appendCommaWhenNeeded(literal);
-        append("'");
-        append(literal.toString());
-        append("'");
-        setLastVisited(literal);
+    public void visit(final Not not) {
+        appendKeyword("NOT");
+        startParenthesis();
+        not.getNegated().accept((BooleanExpressionVisitor) this);
```

Comment:
```suggestion
        not.getNegated().accept((BooleanExpressionVisitor) this);
```
I would really prefer `getOperand()` here. `getNegated` implies for me that you get the result of the operator `NOT` being applied to the operand. Which is the exact opposite.